src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -1288,6 +1293,83 @@ void SolverInterfaceImpl::readScalarData(
   PRECICE_DEBUG("Read value = " << value);
 }
 
+
+int SolverInterfaceImpl::setBoundingBox(
+    const int     meshID,
+    const double *boundingBox) const
+{
+  PRECICE_TRACE(meshID);
+  PRECICE_ASSERT(boundingBox != nullptr);
+  PRECICE_REQUIRE_MESH_MODIFY(meshID);
+
+  // Get the related mesh
+  MeshContext & context = _accessor->meshContext(meshID);
+  mesh::PtrMesh mesh(context.mesh);
+  PRECICE_DEBUG("Define bounding box");
+  // Transform bounds into a suitable format
+  int                 dim = mesh->getDimensions();
+  std::vector<double> bounds(dim * 2);
+
+  for (int d = 0; d < dim; ++d) {
+    // Assert that min is lower or equal to max
+    PRECICE_ASSERT(boundingBox[2 * d] <= boundingBox[2 * d + 1]);
+    bounds[2 * d]     = boundingBox[2 * d];
+    bounds[2 * d + 1] = boundingBox[2 * d + 1];
+  }
```

Comment:
The assertion should be a method (or side condition) of `BoundingBox`.
The copying could be simply a vector constructor.
We could also add an additional constructor to `BoundingBox`, which takes a `double*` instead. 
Let's worry about this later.

Reply:
Alright.